test/access/AccessControl.behavior.js

```diff
@@ -210,8 +211,293 @@ function shouldBehaveLikeAccessControlEnumerable(errorPrefix, admin, authorized,
   });
 }
 
+function shouldBehaveLikeAccessControlDefaultAdminRules(errorPrefix, delay, defaultAdmin, newDefaultAdmin, other) {
+  shouldSupportInterfaces(['AccessControlDefaultAdminRules']);
+
+  it('has a default disabled delayed until', async function () {
+    expect(await this.accessControl.defaultAdminTransferDelayedUntil()).to.be.bignumber.equal(web3.utils.toBN(0));
+  });
+
+  it('has a default pending default admin', async function () {
+    expect(await this.accessControl.pendingDefaultAdmin()).to.equal(ZERO_ADDRESS);
+  });
+
+  it('has a default current owner set to the initial default admin', async function () {
+    const owner = await this.accessControl.owner();
+    expect(owner).to.equal(defaultAdmin);
+    expect(await this.accessControl.hasRole(DEFAULT_ADMIN_ROLE, owner)).to.be.true;
+  });
+
+  it('should revert if granting default admin role', async function () {
+    await expectRevert(
+      this.accessControl.grantRole(DEFAULT_ADMIN_ROLE, defaultAdmin, { from: defaultAdmin }),
+      `${errorPrefix}: can't directly grant defaultAdmin role`,
+    );
+  });
+
+  it('should revert if revoking defaultAdmin role', async function () {
+    await expectRevert(
+      this.accessControl.revokeRole(DEFAULT_ADMIN_ROLE, defaultAdmin, { from: defaultAdmin }),
+      `${errorPrefix}: can't directly revoke defaultAdmin role`,
+    );
+  });
+
+  it("should revert if defaultAdmin's admin is changed", async function () {
+    await expectRevert(
+      this.accessControl.$_setRoleAdmin(DEFAULT_ADMIN_ROLE, defaultAdmin),
+      `${errorPrefix}: can't violate defaultAdmin rules`,
+    );
+  });
+
+  it('should not grant the default admin role twice', async function () {
+    await expectRevert(
+      this.accessControl.$_grantRole(DEFAULT_ADMIN_ROLE, defaultAdmin),
+      `${errorPrefix}: defaultAdmin already granted`,
+    );
+  });
+
+  describe('begins transfer default admin', async function () {
```

Comment:
This should not be async! Please review all `describe`s. (I don't think it has any effect though.)

```suggestion
  describe('begins transfer default admin', function () {
```

Reply:
Any reasons to be strict with this? I was looking for a linter rule but only [found for Typescript](https://typescript-eslint.io/rules/no-misused-promises/) if it helps.

Fixed in https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4009/commits/5a31431a59746687d655ff35ab0264e533513380

Reply:
As far as I know, unless we use `await`, there should be no bad consequence from using async functions here.